rpcwebsocket.go

```diff
@@ -38,6 +38,15 @@ const (
 	websocketSendBufferSize = 50
 )
 
+type semaphore chan struct{}
```

Comment:
Would a `// http://lmgtfy.com/?q=semaphore` work? :p

These kind of useless comments make me want to just remove the type instead and deal with a normal `chan struct{}`.  The type name and method names already document its intended usage.